client/reader/controller.js

```diff
@@ -55,6 +57,23 @@ function trackScrollPage( path, title, category, readerView, pageNum ) {
 	} );
 }
 
+function onRouteChange( newContext, oldContext ) {
+	if ( ! oldContext ) {
+		return;
+	}
+
+	const fullPostViewPrefix = '/read/post/';
+
+	if ( startsWith( oldContext.path, fullPostViewPrefix ) &&
+		! startsWith( newContext.path, fullPostViewPrefix ) &&
+		__fullPostInstance ) {
+		__fullPostInstance.setState( { isVisible: false } );
+		__fullPostInstance = false;
+	}
+}
+
+routeChange( onRouteChange );
```

Comment:
This seems like we're working around the language instead of within in, since we don't appear to use `onRouteChange` anywhere else. Why not just create it inside the call to `routeChange()`?

```js
routeChange( ( newContext, oldContext ) => {
	…
} );
```